pymatgen/io/openmm/utils.py

```diff
@@ -360,111 +360,182 @@ def parameterize_system(
+                temp_ff_string = "charmm"
+            elif "amoeba" in ff.lower():
+                temp_ff_string = "amoeba"
+            else:
+                continue
+            if large_ff_category == "":
+                large_ff_category = temp_ff_string
+            else:
+                if large_ff_category != temp_ff_string:
+                    warnings.warn(f"Did you mean to mix {temp_ff_string} and " f"{large_ff_category} force fields?")
+
+        ff_to_load = ""
         for ff in large_or_water.values():
-            forcefield_omm.loadFile(ff)
-        for mol, ff in small_molecules.items():
-            if "gaff" in ff:
+            if ff in basic_water_ffs:
+                # Ensure the water model matches the large molecule model
+                if large_ff_category:
+                    if large_ff_category == "amber":
+                        if ff == "spce":
+                            ff_to_load = "amber14/spce.xml"
+                        if ff == "tip3p":
+                            ff_to_load = "amber14/tip3p.xml"
+                        if ff == "tip4p":
+                            ff_to_load = "amber14/tip4pew.xml"
+                    elif large_ff_category == "charmm":
+                        if ff == "spce":
+                            ff_to_load = "charmm36/spce.xml"
+                        if ff == "tip3p":
+                            ff_to_load = "charmm36/water.xml"
+                        if ff == "tip4p":
+                            ff_to_load = "charmm36/tip4pew.xml"
+                    else:
+                        warnings.warn(f"Did you mean to use {ff} with the " f"{large_ff_category} force field?")
+                # If there isn't a large molecule forcefield required,
+                # assume amber14
+                else:
+                    if ff == "spce":
+                        ff_to_load = "amber14/spce.xml"
+                    if ff == "tip3p":
+                        ff_to_load = "amber14/tip3p.xml"
+                    if ff == "tip4p":
+                        ff_to_load = "amber14/tip4pew.xml"
+            # TODO: Add lookup to ensure the ff is allowable
+            else:
+                ff_to_load == ff
+            forcefield_omm.loadFile(ff_to_load)
+        # Add small molecules to forcefield
+        for mol, ff_name in small_molecules.items():
+            if "gaff" in ff_name.lower():
+                if ff_name.lower() == "gaff":
+                    ff = "gaff-2.11"
+                else:
+                    ff = ff_name
                 gaff = GAFFTemplateGenerator(molecules=mol, forcefield=ff)
                 forcefield_omm.registerTemplateGenerator(gaff.generator)
-            elif "smirnoff" in ff or "openff" in ff:
+            elif "smirnoff" in ff_name or "openff" in ff_name or "sage" in ff_name:
+                if ff_name.lower() == "sage":
+                    ff = "openff-2.0.0"
+                else:
+                    ff = ff_name
```

Comment:
maybe a case for using a `assign_small_molecule_ff` function

Reply:
Added `assign_small_molecule_ff`